modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/index.d.ts

```diff
@@ -699,3 +699,29 @@ export function toggleSelectBox(
 	value: any,
 	toggleBoxId: string
 ): void;
+
+/* Returns the stored value of a cookie, undefined if not present */
+export function getCookie(name: string): string | undefined;
+
+/* Sets a cookie of a specific type if user has consented */
+export function setCookie(
+	name: string,
+	value: string,
+	type: string,
+	options: {
+		'domain'?: string;
+		'expires'?: string;
+		'max-age'?: string;
+		'path'?: string;
+		'samesite'?: string;
+		'secure'?: boolean;
+	}
+): boolean;
+
+/* Removes a cookie by expiring it */
+export function removeCookie(name: string): void;
+
+/**
+ * Object with cookie consent types as keys, for use with {@link setCookie}
+ */
+export const COOKIE_TYPES: Object;
```

Comment:
@kresimir-coko In TS this would be an enum, and `setCookie` should receive one of the enum values in the `type` param, but since this is just types I didn't want to basically copy the JS object here. Any ideas on how to get this properly typed without duplicating code?

Reply:
As far as the enum type is concerned I haven't had any interactions with them. You should probably be able to find whatever you need in the [official docs](https://www.typescriptlang.org/docs/handbook/enums.html).

I can dig deeper if you want me to, but I'll let you try it out first 😄